### PR TITLE
check NewProxier error

### PR DESCRIPTION
**Pull request by 249043822**
while  NewProxier return error, then the returned proxier will be nil, iptInterface.AddReloadFunc(proxier.Sync) will cause nil exception, so here should return while encounted NewProxier error

### CODE REVIEW

File: cmd/kube-proxy/app/server.go [hunk 1]
```diff
@@ -237,6 +237,9 @@ func NewProxyServerDefault(config *options.ProxyServerConfig) (*ProxyServer, err
 		glog.V(0).Info("Tearing down pure-iptables proxy rules.")
 		iptables.CleanupLeftovers(iptInterface)
 	}
+	if err != nil && proxier == nil {
```
Comment (dims) [inline cmd/kube-proxy/app/server.go]: should this be `err != nil || proxier == nil` since line 243 will fail if proxier is nil? (even if err is nil, proxier should not be nil)

Comment (249043822) [inline cmd/kube-proxy/app/server.go]: hi @thockin , you see functions below:
 iptables.NewProxier(iptInterface, execer, config.IPTablesSyncPeriod.Duration, config.MasqueradeAll, int(*config.IPTablesMasqueradeBit), config.ClusterCIDR)
and 
userspace.NewProxier(
			loadBalancer,
			net.ParseIP(config.BindAddress),
			iptInterface,
			*utilnet.ParsePortRangeOrDie(config.PortRange),
			config.IPTablesSyncPeriod.Duration,
			config.UDPIdleTimeout.Duration,
		)
i think only err !=null && proxier ==nil will appear, err is nil and proxier is still nil could not appear from returns of each NewProxier method above

Comment (thockin) [inline cmd/kube-proxy/app/server.go]: I'm confused.  This err is outside the scope of *.NewProxier() calls, and it was previously checked to be not nil (lines 187-188), so it is guaranteed non-nil here.  If proxier is nil, that would be pretty bad.  Did you actually see this fire?

Comment (thockin) [inline cmd/kube-proxy/app/server.go]: My point is mostly that if err was ever nil, we should have hit a fatal error already.  If we got here with a nil proxier, someone returned no error and a nil value.  I would be fine with a check for nil value here and a fmt.Errorf() explaining it, but this && can never ever trigger, unless I am missing somethign...

Comment (249043822) [inline cmd/kube-proxy/app/server.go]: Sorry @thockin , i had a mistake on the err variable, now i have changed the code, please review again, thanks

Comment (thockin) [inline cmd/kube-proxy/app/server.go]: glog.Fatalf terminates the program.  Your errInf assignments can never
trigger.  I get the sense that you're solving a hypothetical problem. Do
you have a real issue that is triggering?

Comment (249043822) [inline cmd/kube-proxy/app/server.go]: Oh, i didn't know glog.Fatalf terminates the program, so the pr do not need, i close it, thanks
